src/wms/src/main/java/org/geoserver/wms/legendgraphic/FeatureCountProcessor.java

```diff
@@ -231,9 +241,9 @@ public FeatureCountProcessor(GetLegendGraphicRequest request) {
             TargetLabelUpdater duplicatingVisitor = new TargetLabelUpdater(label);
```

Comment:
Do you have to clone the rule, or would a reference be ok? Probably doesn't matter but for complex rules this could take a while

Reply:
I doubt it would make a significant difference, to get the count we had to do a data access and process all the rules (that seems at least an order of magnitude more expensive, or more).